packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/valueRendering.tests.js

```diff
@@ -651,18 +651,75 @@ export default function() {
-                },
-            }).dxHtmlEditor('instance');
+        testModule('init', () => {
+            [null, undefined].forEach(value => {
+                test(`The content should be correct if initial value is ${value}`, function(assert) {
+                    const instance = $('#htmlEditor').dxHtmlEditor({
+                        converter: {
+                            toHtml: (e) => e,
+                        },
+                        value,
+                    }).dxHtmlEditor('instance');
+
+                    const markup = instance.$element().find(`.${CONTENT_CLASS}`).html();
+
+                    assert.strictEqual(markup, '<p><br></p>');
+                });
+            });
 
-            instance.option('value', 'new value');
+            test('The content should be converted correctly', function(assert) {
+                const instance = $('#htmlEditor').dxHtmlEditor({
+                    converter: {
+                        toHtml: () => '<h1>Html converted</h1><p>value</p>',
+                    },
+                    value: 'not html value',
+                }).dxHtmlEditor('instance');
 
-            const markup = instance.$element().find(`.${CONTENT_CLASS}`).html();
+                const markup = instance.$element().find(`.${CONTENT_CLASS}`).html();
 
-            assert.strictEqual(markup, '<h1>Hi!</h1><p>Test</p>');
+                assert.strictEqual(markup, '<h1>Html converted</h1><p>value</p>');
+            });
+
+            [
+                { returnedValue: { value: 'value' }, expectedMarkup: '<p>[object Object]</p>' },
+                { returnedValue: ['string', { object: 'object' }], expectedMarkup: '<p>string,[object Object]</p>' },
+                { returnedValue: NaN, expectedMarkup: '<p><br></p>' },
+                { returnedValue: true, expectedMarkup: '<p>true</p>' },
+                { returnedValue: false, expectedMarkup: '<p><br></p>' },
+                { returnedValue: null, expectedMarkup: '<p><br></p>' },
+                { returnedValue: undefined, expectedMarkup: '<p><br></p>' },
+                { returnedValue: 4, expectedMarkup: '<p>4</p>' },
+                { returnedValue: Infinity, expectedMarkup: '<p>Infinity</p>' },
+                { returnedValue: -Infinity, expectedMarkup: '<p>-Infinity</p>' },
+            ].forEach(({ returnedValue, expectedMarkup }) => {
+                test(`The content should be handled correctly if toHtml returns an unexpected value: ${returnedValue}`, function(assert) {
+                    const instance = $('#htmlEditor').dxHtmlEditor({
+                        converter: {
+                            toHtml: () => returnedValue,
+                        },
+                    }).dxHtmlEditor('instance');
+
+                    const markup = instance.$element().find(`.${CONTENT_CLASS}`).html();
+
+                    assert.strictEqual(markup, expectedMarkup);
+                });
+            });
+        });
```

Comment:
What's the point of these tests?

Reply:
[7894c038d9dd84dd1831695700bdd6ed766c9ec3]